samples/bluetooth/hci_pwr_ctrl/src/main.c

```diff
@@ -0,0 +1,318 @@
+	}
+
+	net_buf_unref(rsp);
+}
+
+static void get_tx_power(u8_t handle_type, u16_t handle, s8_t *tx_pwr_lvl)
+{
+	struct bt_hci_cp_vs_read_tx_power_level *cp;
+	struct bt_hci_rp_vs_read_tx_power_level *rp;
+	struct net_buf *buf, *rsp;
+	int err;
+
+	*tx_pwr_lvl = 0xFF;
+	buf = bt_hci_cmd_create(BT_HCI_OP_VS_READ_TX_POWER_LEVEL,
+				sizeof(*cp));
+	if (!buf) {
+		printk("Unable to allocate command buffer\n");
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(handle);
+	cp->handle_type = handle_type;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_VS_READ_TX_POWER_LEVEL,
+				   buf, &rsp);
+	if (err) {
+		printk("TXP GET HCI err: %d\n", err);
+		return;
+	}
+
+	rp = (void *)rsp->data;
+	if (rp->status) {
+		printk("HCI cmd err: %d\n", rp->status);
+	} else {
+		*tx_pwr_lvl = rp->tx_power_level;
+	}
+
+	net_buf_unref(rsp);
+}
+
+static void connected(struct bt_conn *conn, u8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	s8_t txp;
+	int ret;
+
+	if (err) {
+		printk("Connection failed (err 0x%02x)\n", err);
+	} else {
+		default_conn = bt_conn_ref(conn);
+		ret = bt_hci_get_conn_handle(default_conn,
+					     &default_conn_handle);
+		if (ret) {
+			printk("No connection handle (err %d)\n", ret);
+		} else {
+			/* Send first at the default selected power */
+			bt_addr_le_to_str(bt_conn_get_dst(conn),
+							  addr, sizeof(addr));
+			printk("Connected via connection (%d) at %s\n",
+			       default_conn_handle, addr);
+			get_tx_power(BT_HCI_VS_LL_TX_POWER_CONTROL_CONN,
+				     default_conn_handle, &txp);
+			printk("Connection (%d) - Initial Tx Power = %d\n",
+			       default_conn_handle, txp);
+
+			set_tx_power(BT_HCI_VS_LL_TX_POWER_CONTROL_CONN,
+				     default_conn_handle,
+				     BT_HCI_VS_LL_TX_POWER_LEVEL_NO_PREF);
+			get_tx_power(BT_HCI_VS_LL_TX_POWER_CONTROL_CONN,
+				     default_conn_handle, &txp);
+			printk("Connection (%d) - Tx Power = %d\n",
+			       default_conn_handle, txp);
+		}
+	}
+}
+
+static void disconnected(struct bt_conn *conn, u8_t reason)
+{
+	printk("Disconnected (reason 0x%02x)\n", reason);
+
+	if (default_conn) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+	}
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void bt_ready(int err)
+{
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Bluetooth initialized\n");
+
+	/* Start advertising */
+	err = bt_le_adv_start(param, ad, ARRAY_SIZE(ad),
+			      NULL, 0);
+	if (err) {
+		printk("Advertising failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Dynamic Tx power Beacon started\n");
+}
+
+static void hrs_notify(void)
+{
+	static u8_t heartrate = 90U;
+
+	/* Heartrate measurements simulation */
+	heartrate++;
+	if (heartrate == 160U) {
+		heartrate = 90U;
+	}
+
+	bt_gatt_hrs_notify(heartrate);
+}
+
+void modulate_adv_tx_power(void *p1, void *p2, void *p3)
```

Comment:
nitpick: it modulates for adv and connections, maybe remove _adv_?

Reply:
[Solved](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/787709a21828771d56ba998996e9a82646846617#diff-b9bd630d9245e9d01d535cbaa393a233R235)

Reply:
[Fixed](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/787709a21828771d56ba998996e9a82646846617#diff-b9bd630d9245e9d01d535cbaa393a233R235)